src/App.tsx

```diff
@@ -86,7 +262,14 @@ export const Main = () => {
             <button
               className="bg-white text-dark-body text-sm px-4 py-2 border-light-body border-2 border-solid rounded-xl"
               onClick={() => {
-                console.log("Update");
+
+                //Function for post Update it toggles the modal and set isUpdating to true and see if there is any selected items else it does nothing
+                const selectedPost = posts.find((post) => post.isSelected);
```

Comment:
Same question here, how can we avoid iterating over the whole array on every click?